liota/core/offline_database.py

```diff
@@ -0,0 +1,152 @@
+#                                                                             #
+#  Redistribution and use in source and binary forms, with or without         #
+#  modification, are permitted provided that the following conditions are met:#
+#                                                                             #
+#  - Redistributions of source code must retain the above copyright notice,   #
+#      this list of conditions and the following disclaimer.                  #
+#                                                                             #
+#  - Redistributions in binary form must reproduce the above copyright        #
+#      notice, this list of conditions and the following disclaimer in the    #
+#      documentation and/or other materials provided with the distribution.   #
+#                                                                             #
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"#
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE  #
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE #
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE  #
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR        #
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF       #
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS   #
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+
+import logging
+import sqlite3
+import threading
+import time
+from liota.dcc_comms.dcc_comms import DCCComms
+from liota.dcc_comms.check_connection import CheckConnection
+
+log = logging.getLogger(__name__)
+
+class OfflineDatabase:
+	def __init__(self, table_name, comms, conn=None, data_drain_size=1, draining_frequency=0):
+		"""
+			:param table_name: table_name in which message will be stored
+			:param comms: comms instance of DCCComms
+			:param data_drain_size: how many messages will be drained within each draining_frequency secs defined.
+			:param draining_frequency: frequency with which data will be published after internet connectivity established(like seconds).
+		"""
+		if not isinstance(table_name, basestring):
+			log.error("Table name should be a string.")
+			raise TypeError("Table name should be a string.")
+		if not isinstance(comms, DCCComms):
+			log.error("DCCComms object is expected.")
+			raise TypeError("DCCComms object is expected.")
+		if not isinstance(draining_frequency, float) and not isinstance(draining_frequency, int):
```

Comment:
Python has Number type.  Also, combine draining_frequency check in this if condition check itself.